File: backend/internal/domain/pasien.go
package domain

import "time"

type Pasien struct {
	PasienID        int        `gorm:"primaryKey;column:pasien_id"`
	Email           string     `gorm:"column:email;uniqueIndex;not null"`
	Password        string     `gorm:"column:password;not null"`
	NamaLengkap     string     `gorm:"column:nama_lengkap;not null"`
	NIK             string     `gorm:"column:nik;uniqueIndex;not null"`
	TanggalLahir    time.Time  `gorm:"column:tanggal_lahir;not null"`
	Telepon         string     `gorm:"column:telepon;not null"`
	JenisKelamin    string     `gorm:"column:jenis_kelamin;not null"`
	Alamat          string     `gorm:"column:alamat;not null"`
	Status          string     `gorm:"column:status;default:'active'"`
	ResetCode       string     `gorm:"column:reset_code;default:null"`
	ResetCodeExpiry *time.Time `gorm:"column:reset_code_expiry;type:timestamp;default:null"`
	CreatedAt       time.Time  `gorm:"autoCreateTime:milli;column:created_at"`
	UpdatedAt       time.Time  `gorm:"autoUpdateTime:milli;column:updated_at"`
}

func (Pasien) TableName() string {
	return "pasien"
}

File: pasien-frontend/lib/models/pasien.dart
class Pasien {
  final int pasienId;
  final String nama;
  final String email;
  final String noTelepon;
  final String noTeleponPendamping;
  final String jenisKelamin;
  final String tanggalLahir;
  final String alamat;

  Pasien({
    required this.pasienId,
    required this.nama,
    required this.email,
    required this.noTelepon,
    required this.noTeleponPendamping,
    required this.jenisKelamin,
    required this.tanggalLahir,
    required this.alamat,
  });

  factory Pasien.fromJson(Map<String, dynamic> json) {
    return Pasien(
      pasienId: json['pasien_id'] ?? 0,
      nama: json['nama'] ?? '',
      email: json['email'] ?? '',
      noTelepon: json['no_telepon'] ?? '',
      noTeleponPendamping: json['no_telepon_pendamping'] ?? '',
      jenisKelamin: json['jenis_kelamin'] ?? '',
      tanggalLahir: json['tanggal_lahir'] ?? '',
      alamat: json['alamat'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'pasien_id': pasienId,
      'nama': nama,
      'email': email,
      'no_telepon': noTelepon,
      'no_telepon_pendamping': noTeleponPendamping,
      'jenis_kelamin': jenisKelamin,
      'tanggal_lahir': tanggalLahir,
      'alamat': alamat,
    };
  }
}
